### PR TITLE
Configuration for vulnerability detector LRU caches size added

### LINKED ISSUE #20783: Create internal configuration values for vulnerability detector LRU caches
# Description 

It is desired to make the vulnerability detector LRU cache size configurable in the Vulnerability Detector.

Create 2 keys in the internal configuration file:
`vulnerability-detector.osdata_lru_size`
`vulnerability-detector.translation_lru_size`

# DoD

- [x] Implement modification
- [x] Test function

### PR DESCRIPTION
**Pull request by matias-braida**
|Related issue|
|---|
|Closes #20783 |

## Description

This PR adds vulnerability detector LRU cache size configuration values in the internal configuration file.
Some UTs were modified to support new values.

## Tests

- Compilation without warnings in every supported platform
  - [ ] Linux
- [ ] Check some different configuration values

### CODE REVIEW

File: src/wazuh_modules/vulnerability_scanner/src/policyManager/policyManager.hpp [hunk 1]
```diff
@@ -155,6 +155,16 @@ class PolicyManager final : public Singleton<PolicyManager>
             }
         }
 
+        if (!newPolicy.contains("translation_lru_size"))
+        {
+            newPolicy["translation_lru_size"] = 2048;
```
Comment (Dwordcito) [inline src/wazuh_modules/vulnerability_scanner/src/policyManager/policyManager.hpp:160]: constexpr

File: src/wazuh_modules/vulnerability_scanner/src/policyManager/policyManager.hpp [hunk 2]
```diff
@@ -155,6 +155,16 @@ class PolicyManager final : public Singleton<PolicyManager>
...
+        }
+
+        if (!newPolicy.contains("osdata_lru_size"))
+        {
+            newPolicy["osdata_lru_size"] = 1000;
```
Comment (Dwordcito) [inline src/wazuh_modules/vulnerability_scanner/src/policyManager/policyManager.hpp:165]: constexpr

File: src/wazuh_modules/vulnerability_scanner/src/policyManager/policyManager.hpp [hunk 3]
```diff
@@ -515,6 +525,26 @@ class PolicyManager final : public Singleton<PolicyManager>
...
     }
+
+    /**
+     * @brief Get translation LRU size.
+     *
+     * @return uint32_t translation LRU size.
+     */
+    uint32_t getTranslationLRUSize() const
+    {
+        return m_configuration.at("translation_lru_size").get<uint32_t>();
```
Comment (Dwordcito) [inline src/wazuh_modules/vulnerability_scanner/src/policyManager/policyManager.hpp:536]: camelCase

File: src/wazuh_modules/vulnerability_scanner/src/policyManager/policyManager.hpp [hunk 4]
```diff
@@ -515,6 +525,26 @@ class PolicyManager final : public Singleton<PolicyManager>
...
+    }
+
+    /**
+     * @brief Get osdata LRU size.
+     *
+     * @return uint32_t osdata LRU size.
+     */
+    uint32_t getOsDataLRUSize() const
+    {
+        return m_configuration.at("osdata_lru_size").get<uint32_t>();
```
Comment (Dwordcito) [inline src/wazuh_modules/vulnerability_scanner/src/policyManager/policyManager.hpp:546]: camelCase for osdat_lru_size

Comment (matias-braida) [reply src/wazuh_modules/vulnerability_scanner/src/policyManager/policyManager.hpp:546]: fixed

File: etc/internal_options.conf [hunk 5]
```diff
@@ -423,6 +423,9 @@ sca.commands_timeout=30
 auth.timeout_seconds=1
 auth.timeout_microseconds=0
 
+# Vulnerability detector LRUs size
+vulnerability-detector.translation_lru_size=2048
```
Comment (Dwordcito) [inline etc/internal_options.conf:427]:
```suggestion
vulnerability-detection.translation_lru_size=2048
```

File: etc/internal_options.conf [hunk 6]
```diff
@@ -423,6 +423,9 @@ sca.commands_timeout=30
...
+vulnerability-detector.osdata_lru_size=1000
```
Comment (Dwordcito) [inline etc/internal_options.conf:428]:
```suggestion
vulnerability-detection.osdata_lru_size=1000
```

File: etc/internal_options.conf (hunk 5 above)
Comment (matias-braida) [reply etc/internal_options.conf]: fixed

File: etc/internal_options.conf (hunk 6 above)
Comment (matias-braida) [reply etc/internal_options.conf]: fixed

**Review by GabrielEValenzuela — APPROVED** [review]
LGTM! The result of the tests was satisfactory and expected.

Great job Mati!

**Pull request merged**